tests/dualtor/test_ipinip.py

```diff
@@ -158,3 +161,91 @@ def verify_downstream_packet_to_server(ptfadapter, port, exp_pkt):
         testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), encapsulated_packet, count=10)
         time.sleep(2)
         verify_downstream_packet_to_server(ptfadapter, exp_ptf_port_index, exp_pkt)
+
+
+@pytest.fixture
+def setup_uplink(rand_selected_dut, tbinfo):
+    """
+    Function level fixture.
+    1. Only keep 1 uplink up. Shutdown others to force the bounced back traffic is egressed from monitor port of mirror session
+    2. If there are more than 1 member in the LAG, update the LAG to have only one member
+    """
+    mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+    portchannels = mg_facts['minigraph_portchannels'].keys()
+    up_portchannel = random.choice(portchannels)
+    logger.info("Select uplink {} for testing".format(up_portchannel))
+    # Shutdown other uplinks except for the selected one
+    for pc in portchannels:
+        if  pc != up_portchannel:
+            cmd = "config interface shutdown {}".format(pc)
+            rand_selected_dut.shell(cmd)
+    # Update the LAG if it has more than one member
+    pc_members = mg_facts['minigraph_portchannels'][up_portchannel]['members']
+    if len(pc_members) > 1:
+        cmds = [
+            "sonic-db-cli CONFIG_DB hset 'PORTCHANNEL|{}' 'min_links' 1".format(up_portchannel), # Update min_links
+            "config portchannel member del {} {}".format(up_portchannel, pc_members[1]),         # Remove 1 portchannel member
+            "systemctl unmask teamd",                                                            # Unmask the service
+            "systemctl restart teamd"                                                            # Resart teamd
+        ]
+        rand_selected_dut.shell_cmds(cmds=cmds)
+        time.sleep(300)
+    up_member = pc_members[0]
+    
+    yield mg_facts['minigraph_ptf_indices'][up_member]
+
+    # Startup the uplinks that were shutdown
+    for pc in portchannels:
+        if  pc != up_portchannel:
+            cmd = "config interface startup {}".format(pc)
+            rand_selected_dut.shell(cmd)
+    # Restore the LAG
+    if len(pc_members) > 1:
+        cmds = [
+            "sonic-db-cli CONFIG_DB hset 'PORTCHANNEL|{}' 'min_links' 2".format(up_portchannel), # Update min_links
+            "config portchannel member add {} {}".format(up_portchannel, pc_members[1]),         # Add back portchannel member
+            "systemctl unmask teamd",                                                            # Unmask the service
+            "systemctl restart teamd"                                                            # Resart teamd
+        ]
+        rand_selected_dut.shell_cmds(cmds=cmds)
+        time.sleep(300)
```

Comment:
better to have a `wait_util` here to check if port channels are back ready.

Reply:
Good suggestion. Updated.